src/Catalog/Helpers/Db2CatalogProjection.cs

```diff
@@ -19,29 +22,58 @@ public Db2CatalogProjection(PackageContentUriBuilder packageContentUriBuilder)
         }
 
-        public FeedPackageDetails FromDataRecord(IDataRecord dataRecord)
+        public FeedPackageDetails FromDataReader(DbDataReader dataReader)
         {
-            if (dataRecord == null)
+            if (dataReader == null)
             {
-                throw new ArgumentNullException(nameof(dataRecord));
+                throw new ArgumentNullException(nameof(dataReader));
             }
 
-            var packageId = dataRecord[Db2CatalogProjectionColumnNames.PackageId].ToString();
-            var normalizedPackageVersion = dataRecord[Db2CatalogProjectionColumnNames.NormalizedVersion].ToString();
-            var listed = dataRecord.GetBoolean(dataRecord.GetOrdinal(Db2CatalogProjectionColumnNames.Listed));
-            var hideLicenseReport = dataRecord.GetBoolean(dataRecord.GetOrdinal(Db2CatalogProjectionColumnNames.HideLicenseReport));
+            var packageId = dataReader[Db2CatalogProjectionColumnNames.PackageId].ToString();
+            var normalizedPackageVersion = dataReader[Db2CatalogProjectionColumnNames.NormalizedVersion].ToString();
+            var listed = dataReader.GetBoolean(dataReader.GetOrdinal(Db2CatalogProjectionColumnNames.Listed));
+            var hideLicenseReport = dataReader.GetBoolean(dataReader.GetOrdinal(Db2CatalogProjectionColumnNames.HideLicenseReport));
 
             var packageContentUri = _packageContentUriBuilder.Build(packageId, normalizedPackageVersion);
+            var deprecationInfo = ReadDeprecationInfoFromDataRecord(dataReader);
 
             return new FeedPackageDetails(
                 packageContentUri,
-                dataRecord.ReadDateTime(Db2CatalogProjectionColumnNames.Created).ForceUtc(),
-                dataRecord.ReadNullableUtcDateTime(Db2CatalogProjectionColumnNames.LastEdited),
-                listed ? dataRecord.ReadDateTime(Db2CatalogProjectionColumnNames.Published).ForceUtc() : Constants.UnpublishedDate,
+                dataReader.ReadDateTime(Db2CatalogProjectionColumnNames.Created).ForceUtc(),
+                dataReader.ReadNullableUtcDateTime(Db2CatalogProjectionColumnNames.LastEdited),
+                listed ? dataReader.ReadDateTime(Db2CatalogProjectionColumnNames.Published).ForceUtc() : Constants.UnpublishedDate,
                 packageId,
                 normalizedPackageVersion,
-                hideLicenseReport ? null : dataRecord[Db2CatalogProjectionColumnNames.LicenseNames]?.ToString(),
-                hideLicenseReport ? null : dataRecord[Db2CatalogProjectionColumnNames.LicenseReportUrl]?.ToString());
+                hideLicenseReport ? null : dataReader[Db2CatalogProjectionColumnNames.LicenseNames]?.ToString(),
+                hideLicenseReport ? null : dataReader[Db2CatalogProjectionColumnNames.LicenseReportUrl]?.ToString(),
+                deprecationInfo);
+        }
+
+        public FeedPackageDeprecationInfo ReadDeprecationInfoFromDataRecord(DbDataReader dataReader)
```

Comment:
Any tests for this method and its changes?

Reply:
Added in https://github.com/NuGet/NuGet.Services.Metadata/pull/542/commits/8b9758d296cee6dc79a5f80535fab1f2aad67f7c